crates/re_space_view_spatial/src/scene/contexts/depth_offsets.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use nohash_hasher::IntMap;
+use re_components::DrawOrder;
+use re_log_types::{Component, EntityPathHash};
+use re_viewer_context::{ArchetypeDefinition, SceneContextPart};
+
+// TODO(wumpf): In the new traitified system it would be likely more ergonomic to map from darw order to depth offset
+//              This would also yield more consistent/meaningful archetypes and avoid queries where not needed.
```

Comment:
Issue is that we have this weird thing where DrawOrder is really part of invididual archetypes like Point2D but then we only use it in the global query which undermines a few things. So in the world of archetype querying it would make more sense if Point2D still checks for a draw order but then uses the precomputed table here to map that to a DepthOffset.
I'll clarify in the code as well.